src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -0,0 +1,313 @@
+                nodeParameters.ConnectCancellation.ThrowIfCancellationRequested();
+
+                var peersToDiscover = new List<NetworkAddress>();
+                peersToDiscover.AddRange(SelectPeersToConnectTo());
+
+                if (peersToDiscover.Count == 0)
+                {
+                    PopulateTableWithDNSNodes(network, peersToDiscover);
+                    PopulateTableWithHardNodes(network, peersToDiscover);
+                    peersToDiscover = new List<NetworkAddress>(peersToDiscover.OrderBy(a => RandomUtils.GetInt32()));
+                    if (peersToDiscover.Count == 0)
+                        await Task.CompletedTask;
+                }
+
+                try
+                {
+                    var clonedParameters = nodeParameters.Clone();
+
+                    Parallel.ForEach(peersToDiscover, new ParallelOptions()
+                    {
+                        MaxDegreeOfParallelism = 5,
+                        CancellationToken = nodeParameters.ConnectCancellation,
+                    },
+                    peer =>
+                    {
+                        using (var connectTokenSource = CancellationTokenSource.CreateLinkedTokenSource(nodeParameters.ConnectCancellation))
+                        {
+                            connectTokenSource.CancelAfter(TimeSpan.FromSeconds(5));
+
+                            Node node = null;
+
+                            try
+                            {
+                                clonedParameters.ConnectCancellation = connectTokenSource.Token;
+
+                                var addressManagerBehaviour = clonedParameters.TemplateBehaviors.Find<PeerAddressManagerBehaviour>();
+                                clonedParameters.TemplateBehaviors.Clear();
+                                clonedParameters.TemplateBehaviors.Add(addressManagerBehaviour);
+
+                                node = Node.Connect(network, peer.Endpoint, clonedParameters);
+                                node.VersionHandshake(connectTokenSource.Token);
+                                node.MessageReceived += (s, a) =>
+                                {
+                                    if (a.Message.Payload is AddrPayload addressPayload)
+                                        Interlocked.Add(ref peersFound, addressPayload.Addresses.Length);
+                                };
+
+                                node.SendMessageAsync(new GetAddrPayload());
+
+                                connectTokenSource.Token.WaitHandle.WaitOne(2000);
+                            }
+                            catch
+                            {
+                            }
+                            finally
+                            {
+                                node?.DisconnectAsync();
+                            }
+                        }
+                    });
+                }
+                catch (OperationCanceledException)
+                {
```

Comment:
Is this catch really needed?

Reply:
Nope, you are right, this catch is redundant.